setup/setup_all.sh

```diff
@@ -87,27 +87,33 @@ create_terraform_gcs_bucket() {
 # Run terraform to set up all GCP resources. (Setting up GKE by default)
 init_foundation() {
   # Init Terraform
-  cd "$BASE_DIR"/terraform/stages/foundation
+  cd "${BASE_DIR}"/terraform/stages/foundation
   terraform init -reconfigure -backend-config=bucket="${TF_BUCKET_NAME}"
   
   # Enabling GCP services first.
   terraform apply -target=module.project_services -target=module.service_accounts -auto-approve
-  
+
+
+  # Check if firestore database already exists using gcloud command
+  FIRESTORE_INIT=""
+  if [[ $(gcloud alpha firestore databases list | grep -c uid) == 0 ]]; then
```

Comment:
For a newly created project, this is the result with prompt:

```
(.venv) jonchen-macbookpro3:solutions-template jonchen$ $(gcloud alpha firestore databases list | grep -c uid) == 0
API [firestore.googleapis.com] not enabled on project [498093619004]. Would you like to enable and retry (this will take a few minutes)? 
(y/N)?  
```

You can try `gcloud alpha firestore databases list --quiet`, but please be aware that it will show error:
```
(.venv) jonchen-macbookpro3:solutions-template jonchen$ gcloud alpha firestore databases list --quiet
ERROR: (gcloud.alpha.firestore.databases.list) User [admin@jonchen.altostrat.com] does not have permission to access projects instance [jonchen-st-20230420] (or it may not exist): Cloud Firestore API has not been used in project 498093619004 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/firestore.googleapis.com/overview?project=498093619004 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
- '@type': type.googleapis.com/google.rpc.Help
  links:
  - description: Google developers console API activation
    url: https://console.developers.google.com/apis/api/firestore.googleapis.com/overview?project=498093619004
- '@type': type.googleapis.com/google.rpc.ErrorInfo
  domain: googleapis.com
  metadata:
    consumer: projects/498093619004
    service: firestore.googleapis.com
  reason: SERVICE_DISABLED
```

Reply:
Thanks @jonchenn Firestore API is being enabled here:
https://github.com/GoogleCloudPlatform/solutions-template/blob/main/terraform/stages/foundation/main.tf#L31

Reply:
Make sense, thanks for the fix.